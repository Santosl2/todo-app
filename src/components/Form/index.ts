export * from "./Input";
export * from "./MainForm";
export * from "./Button";
