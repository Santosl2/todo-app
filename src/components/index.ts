export * from "./Header";
export * from "./Form";
export * from "./Badge";
export * from "./TextWithBadge";
export * from "./Item";
export * from "./NoResults";
