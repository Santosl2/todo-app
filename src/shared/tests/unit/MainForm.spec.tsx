import * as Redux from "react-redux";

import { MainForm } from "@/components";
import { fireEvent, screen } from "@testing-library/react";

import { renderWithRedux } from "../utils/CustomRender";

jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useDispatch: jest.fn(),
}));

describe("MainForm", () => {
  const mockedDispatch = Redux.useDispatch;

  afterEach(() => {
    (mockedDispatch as jest.Mock).mockClear();
  });

  it("should render correctly", () => {
    renderWithRedux(<MainForm />);

    expect(
      screen.getByPlaceholderText("O que você precisa fazer?")
    ).toBeInTheDocument();
    expect(screen.getByTestId("createBtnTest")).toBeInTheDocument();
  });

  it("should not be able to create a new todo when value is empty ", () => {
    renderWithRedux(<MainForm />);

    const createBtn = screen.getByTestId("createBtnTest");

    expect(createBtn).toHaveAttribute("disabled");
  });

  it("should not be able to create a new todo", () => {
    (mockedDispatch as jest.Mock).mockReturnValue((e: any) => {
      return e;
    });

    renderWithRedux(<MainForm />);

    const inputAddTodo = screen.getByPlaceholderText(
      "O que você precisa fazer?"
    ) as HTMLInputElement;

    fireEvent.change(inputAddTodo, {
      target: { value: "Only a test" },
    });

    const createBtn = screen.getByTestId("createBtnTest");

    expect(createBtn).not.toHaveAttribute("disabled");

    fireEvent.click(createBtn);

    expect(mockedDispatch).toHaveBeenCalledTimes(3);
  });

  it("match snapshot", () => {
    const { container } = renderWithRedux(<MainForm />);

    expect(container).toMatchSnapshot();
  });
});
