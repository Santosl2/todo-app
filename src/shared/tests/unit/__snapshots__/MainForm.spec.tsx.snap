// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainForm match snapshot 1`] = `
<div>
  <form
    class="flex w-full items-center justify-center gap-2"
  >
    <input
      class="h-[3.375rem] w-full resize-none rounded-lg border border-gray-700 bg-gray-500 pl-4 text-gray-100 placeholder:text-gray-300 focus:border-blue-300 focus:outline-none"
      placeholder="O que vocÃª precisa fazer?"
    />
    <button
      class="flex h-[3.25rem] items-center gap-2 rounded-lg bg-blue-500 px-4 text-sm font-bold text-gray-100 transition-colors hover:bg-blue-300 disabled:cursor-not-allowed disabled:bg-gray-300"
      data-testid="createBtnTest"
      disabled=""
      type="submit"
    >
      Criar 
      <svg
        fill="currentColor"
        height="16"
        viewBox="0 0 256 256"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="none"
          height="256"
          width="256"
        />
        <circle
          cx="128"
          cy="128"
          fill="none"
          r="96"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="24"
        />
        <line
          fill="none"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="24"
          x1="88"
          x2="168"
          y1="128"
          y2="128"
        />
        <line
          fill="none"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="24"
          x1="128"
          x2="128"
          y1="88"
          y2="168"
        />
      </svg>
    </button>
  </form>
</div>
`;
